### PR TITLE
KAA-1376 Fixed

### PR DESCRIPTION
**Pull request by ghost**
Create gen_includes folder in build directory and replace gen/kaa_keys_gen.h there

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

**Comment by rasendubi** [comment]
@kaa-leeroy add to whitelist

File: client/client-multi/client-c/CMakeLists.txt [hunk 1]
```diff
@@ -201,11 +201,19 @@ set(KAA_INCLUDE_DIRS
...
+
+# Creation generated include directory
+file(MAKE_DIRECTORY ${GENERATED_INCLUDE_DIR})
+
 # Build unit tests.
 include(${CMAKE_CURRENT_LIST_DIR}/listfiles/UnitTest.cmake)
 
 include_directories(SYSTEM ${KAA_INCLUDE_PATHS})
 
+include_directories(${GENERATED_INCLUDE_DIR})
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt]: Use [`target_include_directories`](https://cmake.org/cmake/help/v3.3/command/target_include_directories.html).

File: client/client-multi/client-c/CMakeLists.txt [hunk 2]
```diff
@@ -201,11 +201,19 @@ set(KAA_INCLUDE_DIRS
         ${KAA_SRC_FOLDER}
         ${KAA_THIRDPARTY_INCLUDE_DIR})
 
+# Generated include directory
+set(GENERATED_INCLUDE_DIR ${PROJECT_BINARY_DIR}/gen_includes)
+
+# Creation generated include directory
+file(MAKE_DIRECTORY ${GENERATED_INCLUDE_DIR})
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt]: Could be done within `externalproject_add_step`

File: client/client-multi/client-c/CMakeLists.txt [hunk 3]
```diff
@@ -201,11 +201,19 @@ set(KAA_INCLUDE_DIRS
         ${KAA_SRC_FOLDER}
         ${KAA_THIRDPARTY_INCLUDE_DIR})
 
+# Generated include directory
+set(GENERATED_INCLUDE_DIR ${PROJECT_BINARY_DIR}/gen_includes)
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt]: Why not just use `${CMAKE_CURRENT_BINARY_DIR}/kaa_keys/`? Then we can drop "Copying generated files" step.

File: client/client-multi/client-c/CMakeLists.txt (hunk 1 above)
Comment (rasendubi) [reply client/client-multi/client-c/CMakeLists.txt]: Should only be done in else-branch of `if(KAA_RUNTIME_KEY_GENERATION)`

**Review by rasendubi — COMMENTED** [review]
Please, avoid creating extra directories and reuse `${CMAKE_CURRENT_BINARY_DIR}/kaa_keys/`.

**Comment by rasendubi** [comment]
Also, please change commit name to include a short summary. Something like `KAA-1376 Don't place keys in source tree` would do.

**Comment by rasendubi** [comment]
Looks ok to me. Squash commits and you're good to go.

**Comment by vadimol** [comment]
@kaa-leeroy retest this please

**Comment by vadimol** [comment]
@kaa-leeroy retest this please

**Comment by vadimol** [comment]
@kaa-leeroy retest this please

**Review by vadimol — APPROVED** [review]

**Pull request merged**